Frontend/app/problem/display/displayProblem.html

```diff
@@ -45,15 +46,31 @@ <h3 class="panel-title">
               {{step.correctStep}}
             </h3>
           </div>
-          <!-- Only show the buttons in the panel for the current step -->
+          <!-- Only show the buttons in the panel for the current step that's not a solution -->
           <div class="panel-body">
-            <button class="btn btn-primary btn-sm" (click)="showNextStep()" *ngIf="goodStepIndex==activeStepCount">
-              Continue to Next Step
+            <button class="btn btn-primary btn-sm" (click)="showNextStep()"
+            *ngIf="goodStepIndex==activeStepCount && goodStepIndex != problemTree.length-1">
+              Correct, go on Mentii
             </button>
-            <!-- Don't show this button if we are on the first good step, that's the problem declaration -->
-            <button class="btn btn-danger btn-sm" (click)="incorrectGoodStep()" *ngIf="goodStepIndex != 0">
-              This Step is Incorrect
+
+            <!-- Only show this button when we've reached a solution -->
+            <button class="btn btn-primary btn-sm" (click)="showNextStep()"
+            *ngIf="goodStepIndex==activeStepCount && goodStepIndex == problemTree.length-1">
+              Confirm solution
+            </button>
+
+            <!-- Don't show this button if we are on the first good step, that's the
+                 problem declaration. This will be shown on the active step -->
+            <button class="btn btn-danger btn-sm" (click)="incorrectGoodStep()"
+            *ngIf="goodStepIndex != 0 && goodStepIndex == activeStepCount">
+              Something here looks wrong Mentii
             </button>
+
+            <!-- Only show this button on the non active steps -->
+            <a type="button" class="lightBadStep btn btn-sm" (click)="incorrectGoodStep()"
+            *ngIf="goodStepIndex != 0 && goodStepIndex != activeStepCount">
+              Something here looks wrong Mentii
```

Comment:
Using `<a>` instead of `<button>` because of some CSS pseudo-classes that apply to links that I couldn't get working for `<button>`. Otherwise after clicking the `<button>` the text turns black and is hard to read